src/dotnet/APIView/APIViewWeb/Languages/CppLanguageService.cs

```diff
@@ -237,6 +235,11 @@ void ProcessNamespaceNode(string nameSpace)
                     {
                         foreach (var member in leafNamespaceNode.inner)
                         {
+                            // Name space has mix of details namespace and classes
+                            // API View should skip those sub details namespaces also
+                            if (member.kind == NamespaceDeclKind)
```

Comment:
How does this know its a Details namespace?
What if it sees
Foo::CurrentNamespace::MyClass
Foo::CurrentNamespace::AnotherNamespace
Foo::CurrentNamespace::AnotherNamespace::AnotherClass

In the above case the continue would skip processing the AnotherNamespace.  In reality we only want to exclude ::Details namespace.  Can you explain the logic on why a leafNamespaceNode has namespaces in it.

Reply:
ProcessNamespace method is to process each namespace level node which processes all it's children(non namespace). All namespaces are added in main map which is used to call ProcessNamespace node. Reason for this skip logic is to handle a scenario like below in which we need to process only non namespace nodes at that level. 
If we take the example you mentioned. map will have two keys:
```Foo::CurrentNamespace```
```Foo::CurrentNamespace::AnotherNamespace```

When ProcessNamespace is called with key ```Foo::CurrentNamespace``` it should not process any child namespace at that level( This not about skipping ```Details``` namespace)

```::Details``` namespaces are already skipped when we identify whether namespace needs to be processed or not at line number 202.

Reply:
Here is API review created for keyvault keys with this change.
https://apiviewstaging.azurewebsites.net/Assemblies/Review/9d30ba83bec246da890edc17d49c8ad8